src/clr-angular/utils/component/is-attribute-set.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+export function isAttributeSet(value: string | boolean | number | object): boolean {
```

Comment:
In this case, it would only be a `string | boolean`.

Reply:
My idea for that function was to be able to have this type of API
```
<component danger> </component>

<component [danger]="displayDanger"></component>
```

I think that the spinner or progress bar will benefit when it comes to changing the styles on the go.

The issue with that was that there is no clean way to use the:

```js
<component danger></component> 
```
For the reason that Angular will return 
```js
this.danger // '' (empty string)
```

You have to make a few checks before you are sure that the value will be `Truthy`
```js
// => value = ''          // true
// => value = 'true'      // true
// => value = 'anything'  // true
// => value = 'false'     // false
```
so value must pass all this checks so we could have the result below.

```js
// All of them must be true
<component danger></component>
<component danger="true"></component>
<component danger="any-string-value-must-be-true"></component>
<component [danger]="shouldDisplayError"></component>

// ALl of them must be false.
<component></component>
<component danger="false"></component>
<component [danger]="iWillBeAlwaysFalse"></component>
```
  
The checks for the `Object | Number` - I Over Engineered it.  

And I think to be more clear I think that the name of the function must be something like that

```js
isBooleanAttributeSet()
```

I don't think that there is a need to check for something else than this

```js
  // for null just return false no need to check anything
  if (value === null) {
    return false;
  }
  if (typeof value === 'string') {
    // Empty string is valid, 'true' as string is also valid
    return value === '' || value.length > 0 && value !== 'false' ;
  }
  // Boolean value will be read as it is, everything else is false
  return typeof value === 'boolean' ? value : false;
```

I'm not sure that even `null` value is possible - but will keep it just to be sure that there will be no corner case. ( I think that `null` is impossible to be returned as a value from Angular.)